src/contracts/pods/EigenPod.sol

```diff
@@ -306,6 +327,8 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
         // ensure that caller has previously enabled restaking by calling `activateRestaking()`
         hasEnabledRestaking
     {
+        // initializes the  "timestampProvenUntil" for batched zk partial withdrawals to the GENESIS_TIME
+        timestampProvenUntil = GENESIS_TIME;
```

Comment:
why is this inside the `verifyWithdrawalCredentials` function?
seems totally broken
probably we should just  use some sane `startTimestamp` as the default, if `timestampProvenUntil == 0` -- it could be set in the EigenPod constructor, and equal to the timestamp of the first EigenPod deployment, perhaps?

Reply:
Several things I was considering here:
- In terms of setting the startTimestamp value, ideally it could be pod creation timestamp but that varies across all pods.  Is it feasible for us to run a script that calls this function on every pod? I was under the assumption that it probably isn't generally speaking (tho we do have 300 or so pods so may not be actually that bad).  Thats also the reason I set it in the verifyWithdrawalCredentials, is because none of the pods have called that function yet on mainnet.   Again the alternative is calling a setter fucntion on every pod.  Maybe that is actually viable

Reply:
100% can't leave it as-is, since it'll keep resetting the value.
I think what I suggested makes more sense, but I'm sure there's other viable options.
In general, I'm not a fan of having to call something for all existing pods, especially if there's a risk of weird ordering problems (e.g. if it's an issue if we *don't* call some special function on some pods, prior to users making a proof request), since that feels like it could necessitate a sequencing of (1) upgrade, (2) call special function, (3) upgrade again -- that's a lot of operational work and review to make sure everything is correct!

Reply:
Yeah this was an attempt to avoid anything involving calling all existing pods.  And yeah blanked on the fact that there are  multiple validators per pod, smh.  What if we did something stupid like this:

if timestampProvenUntil == 0:
        timestampProvenUntil = GENESIS_TIME;

podOwners are forced to call verifyWithdrawalCredentials in order to set themselves as active.  We just need to make sure that any function that timestampProvenUntil is set in, can never be called before verifyWithdrawalCredentials, which seems reasomable

Reply:
Actually, we should just add this to the requestProof function, not verify withdrawal credentials lol